generators/entity/templates/server/src/main/java/package/service/_EntityQueryService.java

```diff
@@ -0,0 +1,163 @@
+<%#
+ Copyright 2013-2017 the original author or authors.
+
+ This file is part of the JHipster project, see https://jhipster.github.io/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%=packageName%>.service;
+<%  const serviceClassName = entityClass + 'QueryService';
+    const instanceType = (dto === 'mapstruct') ? entityClass + 'DTO' : entityClass;
+    const instanceName = (dto === 'mapstruct') ? entityInstance + 'DTO' : entityInstance;
+    const mapper = entityInstance  + 'Mapper';
+    const dtoToEntity = mapper + '.'+ 'toEntity';
+    const entityToDto = mapper + '.'+ 'toDto';
+    const entityListToDto = mapper + '.' + 'toDto';
+    const entityToDtoReference = mapper + '::'+ 'toDto';
+    const repository = entityInstance  + 'Repository';
+    const criteria = entityClass + 'Criteria';%>
+
+<%_ if (fieldsContainLocalDate === true) { _%>
+import java.time.LocalDate;<% } %><% if (fieldsContainZonedDateTime === true) { %>
+import java.time.ZonedDateTime;<% } if (fieldsContainBigDecimal === true) { %>
+import java.math.BigDecimal;<% } %>
+
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specifications;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import io.github.jhipster.service.QueryService;
+
+import <%=packageName%>.domain.<%= entityClass %>;
+import <%=packageName%>.domain.*; // for static metamodels
+import <%=packageName%>.repository.<%= entityClass %>Repository;<% if (searchEngine === 'elasticsearch') { %>
+import <%=packageName%>.repository.search.<%= entityClass %>SearchRepository;<% } %>
+import <%=packageName%>.service.dto.<%= entityClass %>Criteria;
+<% if (dto === 'mapstruct') { %>
+import <%=packageName%>.service.dto.<%= entityClass %>DTO;
+import <%=packageName%>.service.mapper.<%= entityClass %>Mapper;<% } %>
+<%_ for (idx in fields) { if (fields[idx].fieldIsEnum === true) { _%>
+import <%=packageName%>.domain.enumeration.<%= fields[idx].fieldType %>;
+<%_ } } _%>
+<%_
+  function getSpecificationBuilder(fieldType) {
```

Comment:
can you move the functions to `generator-base-private` we need to keep our templates simpler